presto-main/src/main/java/com/facebook/presto/server/QueryStateInfo.java

```diff
@@ -63,34 +68,45 @@ private QueryStateInfo(
         this.user = requireNonNull(user, "user is null");
         this.catalog = requireNonNull(catalog, "catalog is null");
         this.schema = requireNonNull(schema, "schema is null");
-        requireNonNull(resourceGroupChain, "resourceGroupChainInfo is null");
-        this.resourceGroupChain = ImmutableList.copyOf(resourceGroupChain);
+        checkArgument(queryState != QUEUED || resourceGroupChain != null, "Queued query does not have resourceGroupChain");
```

Comment:
Using nulls here makes this complicated. Use `Optional` or an empty list (if that works for this data model).